azurerm/internal/services/hdinsight/hdinsight_kafka_cluster_resource.go

```diff
@@ -100,13 +107,36 @@ func resourceArmHDInsightKafkaCluster() *schema.Resource {
 				MaxItems: 1,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"head_node": SchemaHDInsightNodeDefinition("roles.0.head_node", hdInsightKafkaClusterHeadNodeDefinition),
+						"head_node": SchemaHDInsightNodeDefinition("roles.0.head_node", hdInsightKafkaClusterHeadNodeDefinition, true),
 
-						"worker_node": SchemaHDInsightNodeDefinition("roles.0.worker_node", hdInsightKafkaClusterWorkerNodeDefinition),
+						"worker_node": SchemaHDInsightNodeDefinition("roles.0.worker_node", hdInsightKafkaClusterWorkerNodeDefinition, true),
 
-						"zookeeper_node": SchemaHDInsightNodeDefinition("roles.0.zookeeper_node", hdInsightKafkaClusterZookeeperNodeDefinition),
+						"zookeeper_node": SchemaHDInsightNodeDefinition("roles.0.zookeeper_node", hdInsightKafkaClusterZookeeperNodeDefinition, true),
+
+						"kafka_management_node": SchemaHDInsightNodeDefinition("roles.0.kafka_management_node", hdInsightKafkaClusterKafkaManagementNodeDefinition, false),
+					},
+				},
+			},
+
+			"rest_proxy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"security_group_name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
```

Comment:
API will asks for this property, otherwise it will complain lack of the group name. I think the capacity to look up the group name based on the group id is out of the scope of azurerm provider, it is handled in azuread provider I assume.

I agree it seems unnecessary from the user's perspective, but I don't know how we can improve this given current API behavior.

Reply:
I mean we should file an API bug here, since providing both seems unnecessary - in the interim we can look that up using the Azure AD API's here?